addon/helpers/has-next.js

```diff
@@ -1,13 +1,27 @@
+import { helper } from '@ember/component/helper';
 import { isPresent } from '@ember/utils';
 import { next } from './next';
-import createNeedleHaystackHelper from '../-private/create-needle-haystack-helper';
 import isEqual from '../utils/is-equal';
 
-export function hasNext(currentValue, array, useDeepEqual = false) {
+function hasNext(currentValue, array, useDeepEqual = false) {
   let nextValue = next(currentValue, array, useDeepEqual);
   let isNotSameValue = !isEqual(nextValue, currentValue, useDeepEqual);
 
   return isNotSameValue && isPresent(nextValue);
 }
 
-export default createNeedleHaystackHelper(hasNext);
+export default helper(function(params) {
+  // TODO: GJ: this param munging is currently duplicated
+  let currentValue = params[0];
+
+  let array;
+  let useDeepEqual = false;
+  if (params.length === 2) {
+    array = params[1];
+  } else {
+    useDeepEqual = params[1];
+    array = params[2];
+  }
```

Comment:
`next`, `previous`, `has-next` and `has-previous` seem to have a slightly weird argument ordering where the optional `useDeepEqual` can be in the middle of two required arguments `currentValue` and `array`.

I wonder if we can reorder them as part of the breaking change?

Reply:
that sounds fine to me.  I'll make sure we document this!  Perhaps `warn` if that arg is still a boolean?

Reply:
I've decided to leave the API as is, it seems the defining the `array` in the last position is by design